packages/kotti-ui/source/kotti-filter/KtFilter.vue

```diff
@@ -0,0 +1,219 @@
+<template>
+	<div class="kt-filter">
+		<div class="kt-filter__content">
+			<div
+				v-for="(filter, index) in value"
+				:key="`${filter.key}_${index}`"
+				class="kt-filter__content__filter-row"
+			>
+				<FilterRow
+					:column="getColumn(filter.key)"
+					:columnOptions="getColumnOptions(filter.key)"
+					:filter="filter"
+					:index="index"
+					@remove-click="handleRemoveFilter"
+					@select-column="handleSelectColumn"
+					@select-operation="handleSelectOperation"
+					@set-value="handleSetValue"
+				/>
+			</div>
+			<div v-show="isAddingFilter" class="kt-filter__content__filter-row">
+				<FilterRow
+					:columnOptions="getColumnOptions()"
+					:index="value.length"
+					@remove-click="handleHideAddingFilter"
+					@select-column="handleAddFilter"
+				/>
+			</div>
+		</div>
+		<div class="kt-filter__footer">
+			<FilterButtonLink
+				:icon="Yoco.Icon.PLUS"
+				label="Add filter"
+				@click="handleShowAddingFilter"
+			/>
+			<FilterButtonLink
+				:icon="Yoco.Icon.CLOSE"
+				label="Clear all"
+				@click="handleClear"
+			/>
+		</div>
+	</div>
+</template>
+
+<script lang="ts">
+import { Yoco } from '@3yourmind/yoco'
+import { computed, defineComponent, ref } from '@vue/composition-api'
+
+import { Kotti } from '../types'
+
+import FilterButtonLink from './components/FilterButtonLink.vue'
+import FilterRow from './components/FilterRow.vue'
+import { getEmptyFilter } from './utils'
+
+export default defineComponent<Kotti.Filter.InternalProps>({
+	name: 'KtFilter',
+	props: {
+		columns: {
+			type: Array,
+			default: [],
+		},
+		value: {
+			type: Array,
+			default: [],
+		},
+		isLoading: {
+			type: Boolean,
+			default: false,
+		},
+	},
+	components: {
+		FilterButtonLink,
+		FilterRow,
+	},
+	setup(props, { emit }) {
+		const updateValueCallback = 'update-value'
+		const isAddingFilter = ref<boolean>(false)
+
+		const selectedColumnsKeys = computed<Kotti.Filter.InternalFilter['key'][]>(
+			() =>
+				props.value.map((filter: Kotti.Filter.InternalFilter) => filter.key),
+		)
+
+		const getColumn = (
+			columnKey: Kotti.Filter.InternalColumn['key'],
+		): Kotti.Filter.InternalColumn =>
+			props.columns.find(
+				(column: Kotti.Filter.InternalColumn) => column.key === columnKey,
+			)
+
+		const getColumnOptions = (
+			selectedColumnKey: Kotti.Filter.InternalColumn['key'],
+		): Kotti.FieldSingleSelect.Props['options'] =>
+			props.columns
+				.filter(
+					(column: Kotti.Filter.InternalColumn) =>
```

Comment:
Why is this type annotation necessary? Shouldn’t TypeScript figure it out on its own?